### PR TITLE
Feature: use commands instead of arguments + env0 executable

### PR DESCRIPTION
**Pull request by yaronya**
### Feature Request
1. Actions should be parsed as commands (aka `env0 deploy`) and not argument as it is today (aka `env0 -a Deploy`)
2. env0 cli should be available as an executable and not via `node` as it is today.

### Solution
1.  <br/>  
   * Use `command-line-args` existing features to run `destroy`/`deploy` as commands.
2. <br/>
  * Add a shell-executable `index.js` file that requires the existing `env0-deploy-cli.js` file and runs it's default exported method.
  * Add a `bin` attribute inside `package.json` that references that `index.js` file.

> Note: the executable part was included in this PR since it was difficult to test an immediate called function required from the `env0-deploy-cli.js` module.

### CODE REVIEW

**Comment by yaronya** [comment]
[Make CLI command an argument and not a param](https://trello.com/c/addgzfAT/1644-make-cli-command-an-argument-and-not-a-param)

**Comment by yaronya** [comment]
[Make CLI 'executable'](https://trello.com/c/2HafgEi0/1642-make-cli-executable)

File: node/bin/index.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env node
+
+const run = require('../src/env0-deploy-cli');
+
+run();
```
Comment (avnerenv0) [inline node/bin/index.js:5]: missing newline at the end of the file

File: node/src/env0-deploy-cli.js [hunk 2]
```diff
@@ -33,53 +49,82 @@ const header = `
...
+      await runDeployment(command, commandOptions, environmentVariables);
     }
   } catch (error) {
-    console.log(error);
+    let { message } = error;
+    if (error.response && error.response.data && error.response.data.message) {
+      message += `: ${error.response.data.message}`
+    }
+
+    console.log(message);
```
Comment (avnerenv0) [inline node/src/env0-deploy-cli.js:117]: should go to stderr
```suggestion
    console.error(message);
```

File: node/src/env0-deploy-cli.js [hunk 3]
```diff
@@ -2,22 +2,38 @@ const commandLineArgs = require('command-line-args');
...
+const availableCommands = {
+  'deploy': {
+    options: optionDefinitions,
+    description: 'Deploy an environment'
+  },
+  'destroy': {
+    options: optionDefinitions,
+    description: 'Destroy an environment'
+  },
+  'help': {
```
Comment (avnerenv0) [inline node/src/env0-deploy-cli.js:32]: not sure what you were going for but 
`> env0 help`
will return
`Missing required options: organizationId,projectId,apiKey,apiSecret`

Comment (yaronya) [reply node/src/env0-deploy-cli.js:32]: 26d510d

File: node/bin/index.js (hunk 1 above)
Comment (yaronya) [reply node/bin/index.js]: Actually when I moved `index.js` into `bin/` then it stopped working for me 🤷 
so I moved it back into root: 290005d and added that missing newline 😉

File: node/src/env0-deploy-cli.js (hunk 2 above)
Comment (yaronya) [reply node/src/env0-deploy-cli.js]: 673f335

**Comment by yaronya** [comment]
> Please also notice that publishing to npm from the subfolder might be a little annoying (see https://stackoverflow.com/questions/38935176/how-to-npm-publish-specific-folder-but-as-package-root), and it also means for example that I can't npm install directly from this repo, so maybe we wanna give this guy his own repo or at least make it so the package.json is in the repo root.

@avnerenv0 Roger that. I'll deal with it when I reach the publish task. If I see that it doesn't get along I'd make the required changes 👍

**Pull request merged**